File: thriving-village/backend/src/utils/roster-request.ts
// Shared by academy-cohort (create/list, cohort-scoped), academy-admin
// (list-all), and academy-roster-request (status update) controllers so the
// response shape can't drift between the three places that return one.
export function shapeRosterRequest(r: any) {
  return {
    id: r.id,
    status: r.status,
    count: r.count,
    note: r.note,
    createdAt: r.createdAt,
    cohort: r.cohort ? { id: r.cohort.id, name: r.cohort.name, courseTitle: r.cohort.course?.title } : null,
    facilitator: r.facilitator ? { id: r.facilitator.id, name: r.facilitator.name || r.facilitator.username } : null,
  };
}

File: thriving-village/backend_django/apps/academy/roster_request.py
"""Port of backend/src/utils/roster-request.ts. Shared by the cohort-scoped
create/list actions and the (Stage 11) admin-wide listing so the response
shape can't drift between the places that return one."""


def shape_roster_request(r) -> dict:
    return {
        "id": r.id,
        "status": r.status,
        "count": r.count,
        "note": r.note,
        "createdAt": r.created_at,
        "cohort": {"id": r.cohort.id, "name": r.cohort.name, "courseTitle": r.cohort.course.title} if r.cohort_id else None,
        "facilitator": {"id": r.facilitator.id, "name": r.facilitator.name or r.facilitator.username} if r.facilitator_id else None,
    }
